### DISCUSSION TITLE
Delta failed on push to cloud.

### DISCUSSION THREAD

**Discussion by Rubendbdv** [opengisch/QField#4023] [Bugs]
Greetings,

Past Tuesday March 7 I added points in a project (Inventory of trees on a property), I pushed the local information to the cloud but the next day after synchronising the points dissapeared.

Specifics we use:
-Tablet (Trimble R2) with Qfield (version 2.6.3)
-Desktop (different devices, OS Windows 11) with Qgis 3.22.15
-Connection with WIFI aswell as Mobile Network
-Qfieldcloud username: "thomasdbdv"
-Project: "boombeheer_maasmechelen"

On Tuesday at 3:42 PM I pushed the information of 9 trees to the cloud. When looking at the 'jobs' tab on app.qfield.cloud, we can see that the delta failed, however we never recieved a notification in Qfield that the push failed. The next morning when synchronising the project in Qfield on our tablet, the points from the previous day dissapeared (which is only logical because they were not on the cloud). 
![image](https://user-images.githubusercontent.com/125654399/224244342-1bfa4b63-10dc-4076-b3b8-2bb968d99056.png)

In the 'changes' tab on app.qfield.cloud it shows that from 1:11 PM on March 3rd the changes gave an error message, but it succesfully applied delta (see cursor).
![image](https://user-images.githubusercontent.com/125654399/224246237-3c676c99-965c-4fa0-8469-fa3f2360c7b3.png)

From March 7 at 3:42 PM it does not give the information that delta applied succesfully.

Our storage on the cloud is full, is that the cause of this?

Thanks in advance.

Sincerely,
Ruben

**Comment by suricactus** [comment]
Hey, @Rubendbdv. It is a very interesting case you are describing, congratulations :) .

You can actually see partly what was going on with your change if you click on the `dadef21c` and on the popup you can see "See jobs", there you can find in the logs of job `db54bbdd` that you have:

> 13:24:43.978 QGSMSGLOG WARNING  Picture expression eval error

I guess you have changed the QGIS file with invalid picture expression?

However it is more imporant this error message:

> An unknown error has been encountered while applying delta: 'NoneType' object has no attribute 'strip'.

This prevents your delta to be applied on the second run.

Showing the "error" and "successfully applied changes" is obviously an error from our side, I am adding it to our bug squashing list.

Hope I helped you.

**Comment by StevenSteeno** [comment]
Hello @suricactus, 

I am a collaborator of Ruben in the same organization. 

Thanks for your reply, but unfortunately your answer does not help us at the moment. Could you have a closer look please?

1. We are aware of the expresssion warning, but that is a known problem that has not impacted synchronization of the project in the past.
2. The 'unknown error' - 'NoneType' object has no attribure 'strip' is probably severe, but there is no mention of any specific details of the error (stacktrace or whatever) which makes this error quite impossible to solve for intermediate users. 
Is there any way of getting more information regarding this error? It has started occurring after a qfield cloud 'push changes' action, which somehow possibly corrupted our project in QField Cloud.

![image](https://user-images.githubusercontent.com/127657330/224536259-5fea737c-71c3-4390-835a-5808190abf6b.png)

Please also note that pushing changes from any device is currenly no longer possible on this project in our organisation (for any user). The same errors keep preventing us from uploading new geometries in our project. Synchronizing the entire project is still possible however, but this is not our preferred way of working and we need stable collaboration on this project.

3. The jobs responsible for delta processing always fail and also contain following error: ERROR 1: Invalid index : 15 (repeating from 15 to 38) It is also not clear what triggered this error, but it first popped up after the same push changes action that produced the other errors. 
4. The issue might have been related to a storage quota issue, since we previously had an issue with versioning pictures that quickly used up all our storage quota. 

Please advise us if there is a way to fix these errors easily. In my opinion it seems related to QField Cloud, since in QGIS we have no issues on the last version of the project and everything works fine. (given errors in the QField Cloud jobs are not present in QGIS)

**Comment by suricactus** [comment]
Sorry, GitHub did not show that I mentioned this discussion here https://github.com/opengisch/qfieldcloud/pull/549 .
